
'sun' is considered a forbidden identifier by GCC and therefore it creates
a "fixed" version of openssl/opensslconf.h in its library directory.

This breaks the openssl mediation in OmniOS and also prevents the
compiler from seeing any updated version of that file.

To resolve that, the fix that GCC would use is applied directly to the
include file so that it doesn't see a need to fix it.

diff -wpruN '--exclude=*.orig' a~/crypto/opensslconf.h.in a/crypto/opensslconf.h.in
--- a~/crypto/opensslconf.h.in	1970-01-01 00:00:00
+++ a/crypto/opensslconf.h.in	1970-01-01 00:00:00
@@ -120,7 +120,7 @@
    optimization options.  Older Sparc's work better with only UNROLL, but
    there's no way to tell at compile time what it is you're running on */
  
-#if defined( __sun ) || defined ( sun )		/* Newer Sparc's */
+#if defined( __sun ) || defined ( __sun__ )		/* Newer Sparc's */
 #  define DES_PTR
 #  define DES_RISC1
 #  define DES_UNROLL
